### ISSUE TITLE
Error running Export-M365DSCConfiguration with component SPOUserProfileProperty

### ISSUE THREAD

**Issue by ITAndy23** [microsoft/Microsoft365DSC#2000]
Error running Export-M365DSCConfiguration with component SPOUserProfileProperty:
```Export-M365DSCConfiguration -Components @("SPOUserProfileProperty")```
Authenticate with Global Admin account.
Receive the following error:
```
Connecting to {PnP}...✅
[1/1] Extracting [SPOUserProfileProperty]...Get-MgUser : Authentication needed, call Connect-MgGraph.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.601.1\DSCResources\MSFT_SPOUserProfileProperty\MSFT_SPOUs
erProfileProperty.psm1:305 char:9
+         $instances = Get-MgUser -All:$true
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-MgUser_List1], AuthenticationException
    + FullyQualifiedErrorId : Microsoft.Graph.PowerShell.Cmdlets.GetMgUser_List1
```

**Comment by ITAndy23** [comment]
Hi @andikruger

Thanks for the feedback, I read through:
https://microsoft365dsc.com/user-guide/get-started/authentication-and-permissions/
Ran ```Update-M365DSCAllowedGraphScopes -ResourceNameList SPOUserProfileProperty```
It responded with "scoped updated"
However, i still get an error and I'm GA:
```Connecting to {PnP}...✅
[1/1] Extracting [SPOUserProfileProperty]...Get-MgUser : Insufficient privileges to complete the operation.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.601.1\DSCResources\MSFT_SPOUserProfileProperty\MSFT_SPOUs
erProfileProperty.psm1:305 char:9
+         $instances = Get-MgUser -All:$true
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: ({ ConsistencyLe...ndProperty =  }:<>f__AnonymousType54`9) [Get-MgUser_List 
   1], RestException`1
    + FullyQualifiedErrorId : Authorization_RequestDenied,Microsoft.Graph.PowerShell.Cmdlets.GetMgUser_List1
❌
⌛ Export took {66 seconds}```

**Comment by andikrueger** [comment]
Did you approve the permissions as administrator or in the scope of the user?

**Comment by ykuijs** [comment]
The permissions in the current version of the module are a little outdated. Please try to run this command:
`Connect-MgGraph -Scopes Domain.Read.All,Sites.FullControl.All`
And login with the account you are using in the DSC configuration. Then try again!

**Comment by ykuijs** [comment]
FYI: I am working on a PR that fixes these outdated permissions

**Comment by ITAndy23** [comment]
Hi @ykuijs 
I entered 
``` Connect-MgGraph -Scopes Domain.Read.All,Sites.FullControl.All ```
Output was: 
Welcome To Microsoft Graph!
Next I entered:
``` Export-M365DSCConfiguration -Components @("SPOUserProfileProperty") -Path "C:\temp\M365 DSC" -FileName "$DateTime.ps1" ```
Output was: 
``` Connecting to {PnP}...✅
[1/1] Extracting [SPOUserProfileProperty]...Get-MgUser : Authentication needed, call Connect-MgGraph.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.601.1\DSCResources\MSFT_SPOUserProfileProperty\MSFT_SPOUserProfileProperty.psm1:305 char:9
+         $instances = Get-MgUser -All:$true
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-MgUser_List1], AuthenticationException
    + FullyQualifiedErrorId : Microsoft.Graph.PowerShell.Cmdlets.GetMgUser_List1 ```

**Comment by ITAndy23** [comment]
@andikrueger I'm running the script as Global Admin so I don't think there's any extra permissions I need to grant myself.

**Comment by ykuijs** [comment]
Just troubleshooted this issue and was able to reproduce the issue. It looked like there was some kind of caching issue, so I added an additional check to invalidate the cache if the Graph connection doesn't exist.

**Comment by ykuijs** [comment]
Closed via #2039, will be included in the next release
